CONFIG_OPTIONS.md

```diff
@@ -0,0 +1,575 @@
+
+What port to use on SSH.
+
+Default: 22
+
+
+### `ssh_private_key_filename`
+
+string, optional (required if you are doing an onprem deploy)
+
+This is required if you don't use `key_helper` or `ssh_private_key`.
+
+If `key_helper: true` then this field cannot be supplied.
+
+### `ssh_private_key`
+
+string, optional
+
+If this is not set and you are not using `key_helper`, you must provide `ssh_private_key_filename`.
+
+
+### `ssh_user`
+
+string, optional
+
+Username to log into cluster with. If `provider: onprem` then the host VM configuraiton is known to dcos-launch and this value will be calculated based on `os_name`.
+
+Default: core
+
+### `key_helper`
+
+boolean, optional
+
+Generate private SSH keys for the underlying hosts if `true`. In `platform: aws`, this means the user does not have to supply `KeyName` in the template parameters and dcos-launch will fill it in. Similarly, in `platform: azure`, `sshRSAPublicKey` is populated automatically. In the aws case, this key will be deleted from EC2 when the deployment is deleted with dcos-launch
+
+Default: false
+
+### `zen_helper`
+
+boolean, optional
+
+Only to be used with `provider: aws` and zen templates. If `true`, then the network prerequisites for launching a zen cluster will be provided if missing. The resources potentially covered are: Vpc, InternetGateway, PrivateSubnet, and PublicSubnet. As with `key_helper`, these resources will be deleted if dcos-launch is used for destroying the deployment.
+
+Default: false
+
+### `tags`
+
+dict, optional
+
+Arbitrary tags to set on your deployment.
+
+## Template-based deploy params
+
+These params should be present if you are deploying with a template.
+
+### `template_url`
+
+string, required
+
+URL of template to use. E.g. should resolve to a page with a CloudFormation template if you are deploying on AWS.
+
+### `template_parameters`
+
+dict, required
+
+Parameters that should be injected into the template you are using. Example params for an AWS template:
+
+```
+template_parameters:
+    AdminLocation: 0.0.0.0/0
+    PublicSlaveInstanceCount: 1
+    SlaveInstanceCount: 2
+    DefaultInstanceType: m4.large
+```
+
+## Onprem deploy params
+
+### `deployment_name`
+
+string, required
+
+The name of the cloud resource that will be provided by `dcos-launch`. E.g. if you are deploying with an AWS CloudFormation template, then this will be the name of your stack.
+
+### `platform`
+
+string, required
+
+Allowed: aws, gcp, gce
+
+### `installer_url`
+
+string, required
+
+URL to the DC/OS installer for the version of DC/OS you want to use. For latest stable open source DC/OS, this would be `https://downloads.dcos.io/dcos/stable/dcos_generate_config.sh`.
+
+### `installer_port`
+
+integer, optional
+
+Default: 9000
+
+### `num_private_agents`
+
+integer, optional
+
+How many private agents the cluster shoudl have.
+
+Default: 0
+
+### `num_public_agents`
+
+integer, optional
+
+Default: 0
+
+### `num_masters`
+
+integer, required
```

Comment:
Please add basic description. `Number of DC/OS masters` or something similar.

Reply:
addressed